src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineFlexibleStagingTest.java

```diff
@@ -198,8 +202,98 @@ public void testCopyAppEngineContext_appYamlInAppEngineDirectory() throws Except
 
     List<LogRecord> logs = handler.getLogs();
     Assert.assertEquals(0, logs.size());
-    verify(copyService).copyFileAndReplace(appEngineDirectory.toPath().resolve("app.yaml"),
-        stagingDirectory.toPath().resolve("app.yaml"));
+    verify(copyService).copyFileAndReplace(appEngineDirectory.resolve("app.yaml"),
+        stagingDirectory.resolve("app.yaml"));
+  }
+
+  @Test
+  public void testCopyAppEngineContext_includeCronYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("cron.yaml", true /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_doNotIncludeCronYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("cron.yaml", false /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_includeDosYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("cron.yaml", true /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_doNotIncludeDosYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("cron.yaml", false /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_includeDispatchYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("dispatch.yaml", true /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_doNotIncludeDispatchYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("dispatch.yaml", false /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_includeIndexYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("index.yaml", true /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_doNotIncludeIndexYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("index.yaml", false /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_includeQueueYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("queue.yaml", true /* includeSet */);
+  }
+
+  @Test
+  public void testCopyAppEngineContext_doNotIncludeQueueYaml() throws Exception {
+    verifyOptionalConfigurationFileStaged("queue.yaml", false /* includeSet */);
+  }
+
+  private void verifyOptionalConfigurationFileStaged(String filename, boolean includeSet)
```

Comment:
I did not initially understand the variable name "includeSet". I thought this referred to a java.util.Set or some other kind of set. Maybe just `include`?